sdk/typespec/typespec_client_core/src/async_runtime/web_runtime.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use super::{AsyncRuntime, SpawnedTask, TaskFuture};
+use crate::time::Duration;
+use std::pin::Pin;
+
+/// An [`AsyncRuntime`] using `tokio` based APIs.
+pub(crate) struct WebRuntime;
+
+impl AsyncRuntime for WebRuntime {
+    fn spawn(&self, f: TaskFuture) -> SpawnedTask {
+        Box::pin(async {
+            wasm_bindgen_futures::spawn_local(f);
+            Ok(())
+        })
+    }
+
+    fn sleep(&self, duration: Duration) -> TaskFuture {
+        Box::pin(async move {
+            if let Ok(d) = duration.try_into() {
+                gloo_timers::future::sleep(d).await;
```

Comment:
If we're taking a hard dependency on this, add it to the `Cargo.toml` like you did `wasm-bindgen-futures`. Or is it re-exported or otherwise exposed by `wasm-bindgen-futures`?

Reply:
Both the workspace Cargo.toml and crate Cargo.toml has the gloo-timers defined.

Reply:
I don't see it in the root `Cargo.toml` in `main`. If we reference any crates I want them explicitly in the workspace `Cargo.toml`.